File: java-patterns/src/main/java/behavioral/command/Stock.java
package behavioral.command;

/**
 * Usage: 股票类 <br/>
 * Date: 2023/5/15 10:37 <br/>
 *
 * @author <a href="mailto:hanzhang2566@foxmail.com">hanzhang</a>
 */
public class Stock {
    private final String name;

    private final int quantity;


    public Stock() {
        name = "ABC";
        quantity = 10;
    }

    public void buy() {
        System.out.println("Stock [ Name: " + name + ", Quantity: " + quantity + " ] bought");
    }

    public void sell() {
        System.out.println("Stock [ Name: " + name + ", Quantity:" + quantity + " ]sold ");
    }
}

File: go-patterns/behavioral/command/stock.go
package command

import "fmt"

type Stock struct {
	name     string
	quantity int
}

func (s Stock) Buy() {
	fmt.Printf("name : %s, quantity : %d bought. \n", s.name, s.quantity)
}

func (s Stock) Sell() {
	fmt.Printf("name : %s, quantity : %d sold. \n", s.name, s.quantity)
}
